torch/onnx/_type_utils.py

```diff
@@ -0,0 +1,229 @@
+"""Utilities for converting and operating on ONNX, JIT and torch types."""
+from __future__ import annotations
+
+import enum
+from typing import Dict, Optional, Union
+
+from typing_extensions import Literal
+
+import torch
+from torch._C import _onnx as _C_onnx
+
+ScalarName = Literal[
+    "Byte",
+    "Char",
+    "Double",
+    "Float",
+    "Half",
+    "Int",
+    "Long",
+    "Short",
+    "Bool",
+    "ComplexHalf",
+    "ComplexFloat",
+    "ComplexDouble",
+    "QInt8",
+    "QUInt8",
+    "QInt32",
+    "BFloat16",
+    "Undefined",
+]
+
+TorchName = Literal[
+    "bool",
+    "uint8_t",
+    "int8_t",
+    "double",
+    "float",
+    "half",
+    "int",
+    "int64_t",
+    "int16_t",
+    "complex32",
+    "complex64",
+    "complex128",
+    "qint8",
+    "quint8",
+    "qint32",
+    "bfloat16",
+]
+
+
+class ScalarType(enum.IntEnum):
```

Comment:
Should we make this public?

Reply:
Good idea. Do you think we should expose it in `torch.onnx.ScalarType` or `torch.onnx.type_utils.ScalarType`?

Reply:
What are their pros and cons in your opinion?

Reply:
`torch.onnx.ScalarType` is less subject to change because `type_utils` is a name that we may want to change when refactoring. For example we may decide that `ScalarType` should live in another file than `type_utils`. Cons would be a more crowded torch.onnx namespace. 

I think `torch.onnx.ScalarType` may serve us better in terms of api stability. But we may need to name it as `TorchScalarType` for example so that ppl don't confuse it as onnx's scalar type. (which is converted with `.onnx()`)

Reply:
`torch.onnx.TorchScalarType` sounds great.

Reply:
Renamed to `torch.onnx.JitScalarType`